src/implementation.nix

```diff
@@ -278,6 +278,42 @@ in {
             (l.mkDefault out.devShell)
         )
         outputsToExport;
+      # numtide devshell integration
+      devshells = l.mkMerge (
+        l.flatten (
+          l.flip l.mapAttrsToList config.nci.crates (
+            crate: cfg:
+            l.optional (cfg.numtideDevshell != null) {
+              ${cfg.numtideDevshell} = {
+                packagesFrom = [ config.nci.toolchains.shell ];
```

Comment:
I'm not sure whether this is good style. If two crates would specify the same `numtideDevshell`, this would get added twice. It's not a problem in practice, although I'm not sure what this package entails. I'm open to suggestions if you consider this is a problem.

Reply:
it's just the rust toolchain with devshell specific additions added (think rustfmt, clippy, rust-src etc), the build toolchain doesnt have those to keep the derivation closure smaller. it should be okay if there are two of these, i don't think it would cause a problem in practice (nix should just realize they are the same anyway), so we can just keep it